improver_tests/generate_ancillaries/test_GenerateClearskySolarRadiation.py

```diff
@@ -121,3 +124,83 @@ def test__initialise_input_cubes(
         GenerateClearskySolarRadiation()._initialise_input_cubes(
             target_grid, None, linke_turbidity_on_alternate_grid
         )
+
+
+def test__get_irradiance_times():
+
+    time = datetime(2022, 1, 1, 00, 00, tzinfo=timezone.utc)
+    accumulation_period = 3  # in hours
+    temporal_spacing = 60  # in mins
+
+    expected_times = [
+        datetime(2021, 12, 31, 21, 00, tzinfo=timezone.utc),
+        datetime(2021, 12, 31, 22, 00, tzinfo=timezone.utc),
+        datetime(2021, 12, 31, 23, 00, tzinfo=timezone.utc),
+        datetime(2022, 1, 1, 00, 00, tzinfo=timezone.utc),
+    ]
+    result = GenerateClearskySolarRadiation()._get_irradiance_times(
+        time, accumulation_period, temporal_spacing
+    )
+    assert np.all(result == expected_times)
+
+    accumulation_period = 1
+    expected_times = [
+        datetime(2021, 12, 31, 23, 00, tzinfo=timezone.utc),
+        datetime(2022, 1, 1, 00, 00, tzinfo=timezone.utc),
+    ]
+    result = GenerateClearskySolarRadiation()._get_irradiance_times(
+        time, accumulation_period, temporal_spacing
+    )
+    assert np.all(result == expected_times)
+
+    misaligned_temporal_spcaing = 19
+    with pytest.raises(ValueError, match="must be integer multiple"):
+        GenerateClearskySolarRadiation()._get_irradiance_times(
+            time, accumulation_period, misaligned_temporal_spcaing
+        )
+
+
+@pytest.mark.parametrize("accumulation_period", [1, 3, 24])
+@pytest.mark.parametrize("surface_altitude_cube", [None, "surface_altitude"])
+@pytest.mark.parametrize("linke_turbidity_cube", [None, "linke_turbidity"])
+def test_process(
+    request,
+    target_grid,
+    accumulation_period,
+    surface_altitude_cube,
+    linke_turbidity_cube,
+):
+    """Test process method returns cubes with correct structure."""
+    time = datetime(2022, 1, 1, 00, 00)
+
+    optional_vars = {}
+    if surface_altitude_cube is not None:
+        optional_vars["surface_altitude"] = request.getfixturevalue(
```

Comment:
The usage of getfixturevalue here seems kind of messy, but I can't think of a good alternative right at the moment. I'll have a think about this and see if I can come up with a neater way to do it.

Reply:
I agree that this is fairly clumsy. One thing I've noticed here is that `test_process` is mainly checking cube properties and that there is no corresponding unit-test for `_create_solar_radiation_cube`. I think I'll rework `test_process` to `test__create_solar_radiation_cube` and then do a separate check the logic around `surface_altitude` is correct.